targets/TARGET_Ambiq_Micro/TARGET_Apollo3/device/serial_api.c

```diff
@@ -126,31 +126,35 @@ void serial_init(serial_t *obj, PinName tx, PinName rx) {
   MBED_ASSERT((int)uart != NC);
   obj->serial.uart_control = &ap3_uart_control[uart];
   obj->serial.uart_control->inst = uart;
-
-  // ensure that HAL queueing is disabled (we want to use the FIFOs directly)
-  obj->serial.uart_control->cfg.pui8RxBuffer = NULL;
-  obj->serial.uart_control->cfg.pui8TxBuffer = NULL;
-  obj->serial.uart_control->cfg.ui32RxBufferSize = 0;
-  obj->serial.uart_control->cfg.ui32TxBufferSize = 0;
-
-  // memset(obj->serial.uart_control->cfg, 0x00, sizeof(am_hal_uart_config_t)); // ensure config begins zeroed
-
-  // config uart pins
+    // config uart pins
   pinmap_config(tx, serial_tx_pinmap());
   pinmap_config(rx, serial_rx_pinmap());
 
-  // start UART instance
-  MBED_ASSERT(am_hal_uart_initialize(uart, &(obj->serial.uart_control->handle)) == AM_HAL_STATUS_SUCCESS);
-  MBED_ASSERT(am_hal_uart_power_control(obj->serial.uart_control->handle, AM_HAL_SYSCTRL_WAKE, false) == AM_HAL_STATUS_SUCCESS);
-  MBED_ASSERT(am_hal_uart_configure(obj->serial.uart_control->handle, &(obj->serial.uart_control->cfg)) == AM_HAL_STATUS_SUCCESS);
-
-  // set default baud rate and format
-  serial_baud(obj, 9600);
-  serial_format(obj, 8, ParityNone, 1);
+  if(!obj->serial.uart_control->handle)
```

Comment:
i see you using the handle as an ```initialized``` flag here... just commenting b/c i have recently doubted whether this method worked for the IOM modules